xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateStore.java

```diff
@@ -2064,19 +2064,33 @@ private void releaseAllLocksForCurrentUser(final XWikiContext ctx)
 
             // the select clause is compulsory to reach the fullName i.e. the page pointed
             Query query = session
-                .createQuery("select backlink.fullName from XWikiLink as backlink where backlink.id.link = :backlink");
-            query.setString("backlink", this.localEntityReferenceSerializer.serialize(documentReference));
+                .createQuery("select backlink.fullName from XWikiLink as backlink where "
+                    + "backlink.id.link = :backlink");
+
+            WikiReference wikiReference = documentReference.getWikiReference();
+            boolean sameWikiContext = (wikiReference == null || wikiReference.equals(context.getWikiReference()));
```

Comment:
wikiReference cannot be null here, it's mandatory in a DocumentReference.

Reply:
I'm always afraid with the possible NPE :) I'll remove it.